grpc/proto/search_get_v1.proto

```diff
@@ -0,0 +1,267 @@
+  oneof test_value{
+    string value_text = 4;
+    int64 value_int = 5;
+    bool value_boolean = 6;
+    float value_number = 7;
+    TextArray value_text_array = 9;
+    IntArray value_int_array = 10;
+    BooleanArray value_boolean_array = 11;
+    NumberArray value_number_array = 12;
+  };
+}
+
+
+message MetadataRequest {
+  bool uuid = 1;
+  bool vector = 2;
+  bool creation_time_unix = 3;
+  bool last_update_time_unix = 4;
+  bool distance = 5;
+  bool certainty = 6;
+  bool score = 7;
+  bool explain_score = 8;
+  bool is_consistent = 9;
+}
+
+
+message PropertiesRequest {
+  repeated string non_ref_properties = 1;
+  repeated RefPropertiesRequest ref_properties = 2;
+}
+
+message Hybrid {
+  string query = 1;
+  repeated string properties = 2;
+  // protolint:disable:next REPEATED_FIELD_NAMES_PLURALIZED
+  repeated float vector = 3;
+  float alpha = 4;
+  enum FusionType {
+    FUSION_TYPE_UNSPECIFIED = 0;
+    FUSION_TYPE_RANKED = 1;
+    FUSION_TYPE_RELATIVE_SCORE = 2;
+  }
+  FusionType fusion_type = 5;
+}
+
+message NearTextSearch {
+  message Move{
+    float force = 1;
+    repeated string concepts = 2;
+    repeated string uuids = 3;
+  }
+
+  // protolint:disable:next REPEATED_FIELD_NAMES_PLURALIZED
+  repeated string query = 1;
+  optional double certainty = 2;
+  optional double distance = 3;
+  optional Move move_to = 4;
+  optional Move move_away = 5;
+};
+
+message NearImageSearch {
+  string image = 1;
+  optional double certainty = 2;
+  optional double distance = 3;
+};
+
+message NearAudioSearch {
+  string audio = 1;
+  optional double certainty = 2;
+  optional double distance = 3;
+};
+
+message NearVideoSearch {
+  string video = 1;
+  optional double certainty = 2;
+  optional double distance = 3;
+};
+
+message BM25 {
+  string query = 1;
+  repeated string properties = 2;
+}
+
+
+message RefPropertiesRequest {
+  string reference_property = 1;
+  PropertiesRequest properties = 2;
+  MetadataRequest metadata = 3;
+  string target_collection = 4;
+}
+
+message NearVector {
+  // protolint:disable:next REPEATED_FIELD_NAMES_PLURALIZED
+  repeated float vector = 1;
+  optional double certainty = 2;
+  optional double distance = 3;
+}
+
+message NearObject {
+  string id = 1;
+  optional double certainty = 2;
+  optional double distance = 3;
+}
+
+message SearchReplyV1 {
+  float took = 1;
+  repeated SearchResult results = 2;
+  string generative_grouped_result = 3;
+  repeated GroupByResult group_by_results = 4;
+}
+
+message GroupByResult {
+  string name = 1;
+  float min_distance = 2;
+  float max_distance = 3;
+  int64 number_of_objects = 4;
+  repeated SearchResult objects = 5;
+}
+
+message SearchResult {
+  PropertiesResult properties = 1;
+  MetadataResult metadata = 2;
+}
+
+message MetadataResult {
+  string id = 1;
+  // protolint:disable:next REPEATED_FIELD_NAMES_PLURALIZED
+  repeated float vector = 2;
+  int64 creation_time_unix = 3;
+  bool creation_time_unix_present = 4;
```

Comment:
cannot we make `creation_time_unix` just optional? we would then not have to introduce `creation_time_unix_present` and other `_present` settings?

Reply:
sadly doesn't work :(

at least for python there is no difference between optional and non-optional fields. Eg you cannot do `if creation_time is not None` to check if a value is present, it is always there and is just initialized with the default value if not given (0 for numbers, "" for strings). 

For the times we could check if the value is >0 to check for its existence, but for the distance we cannot do that as a distance of 0 could be a valid value. I'd rather keep it consistent and have the _present everywhere

Reply:
OK, I understand now, thanks for clarification